charts/grafana/values.yaml

```
@@ -797,9 +797,9 @@ grafana.ini:
   server:
     domain: "{{ if (and .Values.ingress.enabled .Values.ingress.hosts) }}{{ .Values.ingress.hosts | first }}{{ else }}''{{ end }}"
 ## grafana Authentication can be enabled with the following values on grafana.ini
- # server:
-      # The full public facing url you use in browser, used for redirects and emails
- #    root_url:
+ server:
+    # The full public facing url you use in browser, used for redirects and emails
+    root_url: "{{ if (and .Values.ingress.enabled .Values.ingress.hosts) }}{{if (.Values.ingress.tls )}}https://{{ else }}http://{{end}}{{ .Values.ingress.hosts | first }}{{ else }}''{{ end }}"
```

Could you test, if that works, too?


```suggestion
    protocol: '{{ (and .Values.ingress.enabled .Values.ingress.tls) | ternary "https" "http" }}'
    domain: '{{ (and .Values.ingress.enabled .Values.ingress.hosts) | ternary (.Values.ingress.hosts | first) "localhost" }}'
    root_url: '%(protocol)s://%(domain)s{{ if not .Values.ingress.enabled }}:%(http_port)s{{ end }}/'
```